astroquery/esa/hubble/core.py

```diff
@@ -250,13 +260,83 @@ def get_artifact(self, artifact_id, *, filename=None, verbose=False):
         None. It downloads the artifact indicated
         """
 
-        params = {"RETRIEVAL_TYPE": "PRODUCT", "ARTIFACTID": artifact_id, "TAPCLIENT": "ASTROQUERY"}
+        return self.download_file(file=artifact_id, filename=filename, verbose=verbose)
+
+    def get_associated_files(self, observation_id, *, verbose=False):
+        """
+        Retrieves all the files associated to an observation
+
+        Parameters
+        ----------
+        observation_id : string
+            id of the observation to be downloaded, mandatory
+            The identifier of the observation we want to retrieve, regardless
+            of whether it is simple or composite.
+        verbose : bool
+            optional, default 'False'
+            flag to display information about the process
+
+        Returns
+        -------
+        None. The file is associated
+        """
+        query = (f"select art.artifact_id as filename, p.calibration_level, art.archive_class as type, "
+                 f"pg_size_pretty(art.size_uncompr) as size_uncompressed from ehst.artifact art "
+                 f"join ehst.plane p on p.plane_id = art.plane_id where "
+                 f"art.observation_id = '{observation_id}'")
+        return self.query_tap(query=query)
+
+    def download_fits_files(self, observation_id, *, verbose=False):
```

Comment:
Same API considerations, maybe this could be called `download_files`, and having a kwarg for extensions defaulting to `'fits'`? This is really only a discussion item as `download_files` is not really wide-spread, and the currently existing ones in alma and casda are having different signature

Reply:
We wanted a specific method naming FITS, to make it more visible to users.